src/robotFactory.js

```diff
@@ -1,10 +1,73 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = Object.assign({
+      x: 0, y: 0,
+    }, coords);
```

Comment:
Object.assign is redudant
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0
    }
```